client/src/components/Navbar.js

```diff
@@ -58,14 +62,15 @@ const Navbar = ()=>{
     useEffect(() => {
         if(authContext.isAuthenticated && needsFetchingLists){ 
             fetchShLists(shListsContext.dispatchShLists,shListsContext.handleShListsFailure);
-            needsFetchingLists = false;
+			needsFetchingLists = false;
+			props.history.push("/main");
```

Comment:
not sure if we decided to do this or use the <Redirect /> component from react router dom, I know we briefly touched on that on friday, but maybe we should choose one for the whole app and stick with it? The differences seem to be pretty miniscule but I think <Redirect /> is a little more readable, also 
https://stackoverflow.com/questions/46757676/in-react-is-it-always-better-to-render-a-redirect-than-use-this-props-history-pu/46757873#:~:text=push%20as%20Redirect%20uses%20it,differently%20at%20a%20later%20date.

indicates it may have some implications for scaling. Either way is cool for me though!

Reply:
Ohh that's great! As far as I think, it solves the problem for now. We might have some issues during scaling but that is future work lol. Also we can use the useHistory hooks, anyways we are using latest concepts from React. So it seems fine!

Reply:
Yes, we use `<Redirect>` in Routes.js file. Using `history.push` is not consistent. However, the reason why I decided to use `history.push` here is that  I want to redirect user only one time after logging in, and leave the landing page available for browsing when user click top left corner. Redirect is a component, so I considered where to place this component. The logging-in form might not be a good choice because it's unmounted right away after submitting. I also considered placing `<Redirect>` on other permanent component or Routes.js, then I had to set some flags to control rendering `<Redirect>`.  Compared to `<Redirect>`, `history.push` is a function, and it can be disposable and be easily called within `useEffect` without additional flags, so I use it here.

Reply:
Okay great! Good to know!

Reply:
Thank you for the information about Redirect. Since it's not the main feature on this branch, we may change it on latter branch.

Reply:
Great discussion and good point Irene about allowing the user to visit the landing page and the ease of using history.push.